### PR TITLE
UI vision refactor

### PR DESCRIPTION
**Pull request by adeebshihadeh**
Continuation of #2063. This gets UI and SurfaceFlinger down to a bit under 6% of a core each, getting us 9% of a core back. This also fixes the frequent slow frames and improves general responsiveness.

### CODE REVIEW

**Comment by deanlee** [comment]
Happy to see the code gets better!

File: selfdrive/ui/android_ui.cc [hunk 1]
```diff
@@ -243,109 +238,33 @@ int main(int argc, char* argv[]) {
...
-        }
-
-        s->scene.alert_text1 = "TAKE CONTROL IMMEDIATELY";
-        s->scene.alert_text2 = "Controls Unresponsive";
-        s->scene.alert_size = cereal::ControlsState::AlertSize::FULL;
-        update_status(s, STATUS_ALERT);
-      }
-      ui_draw_vision_alert(s, s->scene.alert_size, s->status, s->scene.alert_text1.c_str(), s->scene.alert_text2.c_str());
+    if (!s->awake) {
+      continue;
```
Comment (deanlee) [inline selfdrive/ui/android_ui.cc:250]: How about moving setVolume & ui_set_brightness below here? no need to call them when awake is false.

File: selfdrive/ui/android_ui.cc [hunk 2]
```diff
@@ -214,25 +206,28 @@ int main(int argc, char* argv[]) {
   const int MAX_VOLUME = LEON ? 15 : 12;
   assert(s->sound.init(MIN_VOLUME));
 
-  int draws = 0;
+  uint32_t draws = 0;
```
Comment (deanlee) [inline selfdrive/ui/android_ui.cc:209]: uint64_t?

Comment (adeebshihadeh) [reply selfdrive/ui/android_ui.cc]: replaced with submaster frame

File: selfdrive/ui/android_ui.cc (hunk 1 above)
Comment (adeebshihadeh) [reply selfdrive/ui/android_ui.cc:238]: done

**Comment by deanlee** [comment]
It’s amazing to reduce so many lines of code after this bigger refactor :beer:

File: selfdrive/ui/android_ui.cc [hunk 3]
```diff
@@ -243,109 +226,40 @@ int main(int argc, char* argv[]) {
...
-        // warn on sub 15fps
-        LOGW("slow frame(%d) time: %.2f", draws, u2-u1);
-      }
-      draws++;
-      framebuffer_swap(s->fb);
+    ui_draw(s);
+    double u2 = millis_since_boot();
+    if (!s->scene.frontview && (u2-u1 > 66)) {
+      // warn on sub 15fps
+      LOGW("slow frame(%d) time: %.2f", (s->sm)->frame, u2-u1);
```
Comment (deanlee) [inline selfdrive/ui/android_ui.cc:255]: (%d) should be (%llu)

Comment (deanlee) [reply selfdrive/ui/android_ui.cc:255]: s->sm->frame may be better?

File: selfdrive/ui/paint.cc [hunk 4]
```diff
@@ -844,7 +840,5 @@ void ui_nvg_init(UIState *s) {
   for(int i = 0;i < UI_BUF_COUNT; i++) {
```
Comment (deanlee) [inline selfdrive/ui/paint.cc:840]: Add a space after the semicolon

File: selfdrive/ui/android_ui.cc (hunk 3 above)
Comment (adeebshihadeh) [reply selfdrive/ui/android_ui.cc:255]: what's the difference with the second thing?

File: selfdrive/controls/controlsd.py [hunk 5]
```diff
@@ -94,7 +94,6 @@ def __init__(self, sm=None, pm=None, can_sock=None):
     cp_bytes = self.CP.to_bytes()
     params.put("CarParams", cp_bytes)
     put_nonblocking("CarParamsCache", cp_bytes)
-    put_nonblocking("LongitudinalControl", "1" if self.CP.openpilotLongitudinalControl else "0")
```
Comment (pd0wm) [inline selfdrive/controls/controlsd.py:97]: Should we remove the param from params.py. As far as I can tell it's unused now.

File: selfdrive/ui/sidebar.cc [hunk 6]
```diff
@@ -159,13 +159,12 @@ static void ui_draw_sidebar_panda_metric(UIState *s) {
...
-    ui_draw_sidebar_metric(s, NULL, NULL, 0, 180+158, "CONNECT\nONLINE");
-  } else {
-    ui_draw_sidebar_metric(s, NULL, NULL, 2, 180+158, "CONNECT\nERROR");
-  }
+  static std::map<NetStatus, const char *> connectivity_text = {
+    {NET_ERROR, "CONNECT\nERROR"},
+    {NET_CONNECTED, "CONNECT\nONLINE"},
+    {NET_DISCONNECTED, "CONNECT\nOFFLINE"},
+  };
+  ui_draw_sidebar_metric(s, NULL, NULL, 1, 180+158, connectivity_text[s->scene.athenaStatus]);
```
Comment (pd0wm) [inline selfdrive/ui/sidebar.cc:167]: Need to change severity. (1, 4th argument). Now the color is always yellow.

File: selfdrive/controls/controlsd.py (hunk 5 above)
Comment (adeebshihadeh) [reply selfdrive/controls/controlsd.py:97]: done

File: selfdrive/ui/sidebar.cc (hunk 6 above)
Comment (adeebshihadeh) [reply selfdrive/ui/sidebar.cc]: fixed

**Pull request merged**